### PR TITLE
Sign up password validation (issue #137)

### LINKED ISSUE #137: password can be ""
**Describe the bug**
When i sign up for a new account, I can leave the password field blank and will still sign up successfully. my password will be "".

**To Reproduce**
1. Go to "sign up here"
2. Put in a username
3. Leave passwords empty
4. Press sign up 
5. account is successfully created

**Severity**
Mark with an x.
[]: Minor effect eg. graphical
[x]: Functional error eg. App does not function correctly
[]: Severe eg. Crashing

**Reproducibility**
Mark with an x.
[x]: Consistent
[]: Occasional
[]: Cannot reproduce at the moment eg. unsure

**Expected behavior**
Password should not be allowed

**Conditions**
Any condition

### PR DESCRIPTION
**Pull request by joshiefu**
Closes #137

**Description**

Users can no longer create an account without a password plus additional modifications:
- The sign-up page will check for matching passwords on blur
- If a field is invalid the helper text will clarify the issue

**Testing**

Checked that the sign-up page will not allow users to create an account without passwords or with mismatched passwords.

**Checklist**
- [x] I have tested my change
- [x] Code builds successfully and all tests pass
- [ ] Docs have been added/updated if needed

### CODE REVIEW

File: split-webapp/split/src/pages/SignUp.js [hunk 1]
```diff
@@ -21,28 +21,53 @@ class SignUp extends Component {
...
+    if (this.emptyPassword()) {
+      validation.validPassword = false;
+      this.setState({
+        validation
+      });
+    }
+
+    if (
+      this.validateUsername &&
+      !this.emptyPassword &&
```
Comment (timovv) [inline split-webapp/split/src/pages/SignUp.js:68]: `validateUsername` and `emptyPassword` are functions, I assume you mean to check the result of the functions here?

Comment (timovv) [inline split-webapp/split/src/pages/SignUp.js]: `validateUsername` and `emptyPassword` are functions, looks like you are meaning to check the result of the functions here?

**Review by timovv — CHANGES_REQUESTED** [review]

File: split-webapp/split/src/pages/SignUp.js (hunk 1 above)
Comment (FrankWangma) [reply split-webapp/split/src/pages/SignUp.js]: Both functions return a boolean, so it works, but would it be better and clearer if they compared the results instead?

**Review by FrankWangma — COMMENTED** [review]
Other than what Timo said, it looks good to me! Might want to take a look

**Review by FrankWangma — APPROVED** [review]

File: split-webapp/split/src/pages/SignUp.js (hunk 1 above)
Comment (joshiefu) [reply split-webapp/split/src/pages/SignUp.js]: Sweet thanks. I've made the changes according to your suggestions.

**Review by timovv — APPROVED** [review]

**Pull request merged**